queries/orderbook/prod_batch_rewards_cip20.sql

```diff
@@ -0,0 +1,289 @@
+    FROM
+        settlement_scores ss -- If there are reported scores,
+        -- there will always be a record of auction participants
+        JOIN auction_participation ap ON ss.auction_id = ap.auction_id -- outer joins made in order to capture non-existent settlements.
+        LEFT OUTER JOIN observed_settlements os ON os.auction_id = ss.auction_id
+        LEFT OUTER JOIN batch_protocol_fees bpf ON bpf.tx_hash = os.tx_hash
+),
+reward_per_auction as (
+    SELECT
+        tx_hash,
+        auction_id,
+        settlement_block,
+        block_deadline,
+        solver,
+        execution_cost,
+        surplus,
+        protocol_fee, -- the protocol fee
+        fee - network_fee_correction as network_fee, -- the network fee
+        surplus + protocol_fee + fee - network_fee_correction - reference_score as uncapped_payment_eth,
+        -- Capped Reward = CLAMP_[-E, E + exec_cost](uncapped_reward_eth)
+        LEAST(
+            GREATEST(
+                - {{EPSILON_LOWER}},
+                surplus + protocol_fee + fee - network_fee_correction - reference_score
+            ),
+            {{EPSILON_UPPER}} + execution_cost
+        ) as capped_payment,
+        winning_score,
+        reference_score,
+        participating_solvers as participating_solvers
+    FROM
+        reward_data
+),
+participation_data as (
+    SELECT
+        tx_hash,
+        unnest(participating_solvers) as participant
+    FROM
+        reward_per_auction
+),
+participation_counts as (
+    SELECT
+        participant as solver,
+        count(*) as num_participating_batches
+    FROM
+        participation_data
+    GROUP BY
+        participant
+),
+primary_rewards as (
+    SELECT
+        rpt.solver,
+        SUM(capped_payment) as payment_wei,
+        SUM(execution_cost) as exececution_cost_wei
+    FROM
+        reward_per_auction rpt
+    GROUP BY
+        solver
+),
+protocol_fees as (
+    SELECT
+        solver,
+        SUM(protocol_fee) as protocol_fee_wei
+    FROM
+        reward_per_auction rpt
+    GROUP BY
+        solver
+),
+aggregate_results_temp as (
+    SELECT
+        concat('0x', encode(pc.solver, 'hex')) as solver,
+        coalesce(payment_wei, 0) as payment_eth,
+        coalesce(exececution_cost_wei, 0) as execution_cost_eth,
+        num_participating_batches,
+        coalesce(protocol_fee_wei, 0) as protocol_fee_eth
+    FROM
+        participation_counts pc
+        LEFT OUTER JOIN primary_rewards pr ON pr.solver = pc.solver
+        LEFT OUTER JOIN protocol_fees pf ON pf.solver = pc.solver
+),
+aggregate_results as (
+    SELECT
+        solver,
+        CASE
+            WHEN payment_eth - execution_cost_eth > 0 THEN payment_eth - execution_cost_eth
+            ELSE 0
+        END AS primary_reward_eth,
+        num_participating_batches,
+        protocol_fee_eth,
+        CASE
+            WHEN payment_eth - execution_cost_eth > 0 THEN execution_cost_eth
+            ELSE payment_eth
+        END AS network_fee_eth
```

Comment:
What is the reason for to have two cases here? I would have expected that `payment_eth - execution_cost_eth as primary_reward` and `execution_cost_eth as network_fee_eth` would have been correct as well.

Reply:
yeah i see your point. that's cleaner so i will do that.

Reply:
just pushed your suggestion